### PR TITLE
feat(cli): Improve `vector generate` syntax and formatting

### PR DESCRIPTION
**Pull request by Jeffail**
This PR changes the `vector generate` syntax to allow forward slashes: `foo/bar,baz/boo`, closing https://github.com/timberio/vector/issues/1684. This is backwards compatible as the pipe syntax can still be used.

The output now also adds empty lines between component types to make the formatting nicer:

```toml
data_dir = "/var/lib/vector/"
dns_servers = []

[sources.source0]
max_length = 102400
type = "stdin"

[transforms.transform0]
inputs = ["source0"]
drop_field = true
drop_invalid = false
type = "json_parser"

[sinks.sink0]
healthcheck = true
inputs = ["transform0"]
type = "console"

[sinks.sink0.buffer]
type = "memory"
max_events = 500
when_full = "block"
```

This is the lowest hanging fruit of https://github.com/timberio/vector/issues/1685, we're still missing:

1. Proper ordering of fields (`inputs` first, followed by `type`, and so on)
2. Nicer indentation
3. All required and common fields for all components

But this stuff would require a fair amount of time investment, and I'd like to have a plan/spec for https://github.com/timberio/vector/issues/1735 before we go ahead with it. I'll update #1685 accordingly.

### CODE REVIEW

File: src/generate.rs [hunk 1]
```diff
@@ -199,18 +200,197 @@ pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
...
             exitcode::SOFTWARE
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // !IF YOU MODIFIED COMPONENT FIELDS AND NOW THESE TESTS FAIL!
```
Comment (Hoverbear) [inline src/generate.rs]: I don't really think this is necessary. `generate` is a maturing feature and our team should start to care about it the same way they do other parts of the code. Let folks get annoyed by this, so we get motivated to find a better way. :)

File: src/generate.rs [hunk 2]
```diff
@@ -162,7 +163,7 @@ pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
                 Ok(example) => example,
                 Err(err) => {
                     if err != ExampleError::MissingExample {
-                        errs.push(err);
+                        errs.push(format!("{}", err));
```
Comment (Hoverbear) [inline src/generate.rs]: Are these necessary? Or a leftover?

**Comment by Hoverbear** [comment]
@Jeffail Are docs updates coming in another PR?

**Comment by Hoverbear** [comment]
Looks so much better!

```toml
$ cargo run -- generate stdin,kafka/add_fields,remove_fields,json_parser,json_parser/console
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `target/debug/vector generate stdin,kafka/add_fields,remove_fields,json_parser,json_parser/console`
data_dir = "/var/lib/vector/"
dns_servers = []

[sources.source0]
max_length = 102400
type = "stdin"

[sources.source1]
type = "kafka"

[transforms.transform0]
inputs = ["source0", "source1"]
type = "add_fields"

[transforms.transform1]
inputs = ["transform0"]
type = "remove_fields"

[transforms.transform2]
inputs = ["transform1"]
drop_field = true
drop_invalid = false
type = "json_parser"

[transforms.transform3]
inputs = ["transform2"]
drop_field = true
drop_invalid = false
type = "json_parser"

[sinks.sink0]
healthcheck = true
inputs = ["transform3"]
type = "console"

[sinks.sink0.buffer]
type = "memory"
max_events = 500
when_full = "block"
```

**Comment by Jeffail** [comment]
> @Jeffail Are docs updates coming in another PR?

Not for this one, not sure there's any mention of `generate` on the site yet (won't be the case soon, after my configs guide)

**Comment by Hoverbear** [comment]
Ok! :)

File: src/generate.rs (hunk 1 above)
Comment (Jeffail) [reply src/generate.rs]: True, fair enough.

**Pull request merged**